mutagen/wave.py

```diff
@@ -0,0 +1,369 @@
+    sample_rate = 0
+
+    @convert_error(IOError, error)
+    def __init__(self, fileobj):
+        """Raises error"""
+
+        waveFile = WaveFile(fileobj)
+        try:
+            waveFormatChunk = waveFile[u'fmt ']
+        except KeyError as e:
+            raise error(str(e))
+
+        data = waveFormatChunk.read()
+
+        #  RIFF: http://soundfile.sapp.org/doc/WaveFormat/
+        #  Python struct.unpack:
+        #    https://docs.python.org/2/library/struct.html#byte-order-size-and-alignment
+        info = struct.unpack('<hhLLhh', data[:16])
+        self.audioFormat, self.channels, self.sample_rate, byte_rate, \
+            block_align, self.sample_size = info
+        self.bitrate = self.channels * block_align * self.sample_rate
+
+        try:
+            waveDataChunk = waveFile[u'data']
+        except KeyError as e:
+            raise error(str(e))
+
+        self.number_of_samples = waveDataChunk.data_size / block_align
+        self.length = self.number_of_samples / self.sample_rate
+
+    def pprint(self):
+        return u"%d channel AIFF @ %d bps, %s Hz, %.2f seconds" % (
+            self.channels, self.bitrate, self.sample_rate, self.length)
+
+
+class _WaveID3(ID3):
+    """A Wave file with ID3v2 tags"""
+
+    def _pre_load_header(self, fileobj):
+        try:
+            fileobj.seek(WaveFile(fileobj)[u'id3 '].data_offset)
+        except (InvalidChunk, KeyError):
+            raise ID3NoHeaderError("No ID3 chunk")
+
+    @convert_error(IOError, error)
+    @loadfile(writable=True)
+    def save(self, filething, v2_version=4, v23_sep='/', padding=None):
+        """Save ID3v2 data to the Wave/RIFF file"""
+
+        fileobj = filething.fileobj
+
+        wave_file = WaveFile(fileobj)
+
+        if u'id3 ' not in wave_file:
+            wave_file.insert_chunk(u'id3 ')
+
+        chunk = wave_file[u'id3 ']
+
+        try:
+            data = self._prepare_data(
+                fileobj, chunk.data_offset, chunk.data_size, v2_version,
+                v23_sep, padding)
+        except ID3Error as e:
+            reraise(error, e, sys.exc_info()[2])
+
+        new_size = len(data)
+        new_size += new_size % 2  # pad byte
+        assert new_size % 2 == 0
+        chunk.resize(new_size)
+        data += (new_size - len(data)) * b'\x00'
+        assert new_size == len(data)
+        chunk.write(data)
+
+    @loadfile(writable=True)
+    def delete(self, filething):
+        """Completely removes the ID3 chunk from the RIFF/WAVE file"""
+
+        fileobj = filething.fileobj
+
+        waveFile = WaveFile(fileobj)
+
+        if 'id3 ' in waveFile:
+            waveFile['id3 '].delete()
+        if 'ID3 ' in waveFile:
+            waveFile['ID3 '].delete()
+
+        self.clear()
+
+
+@convert_error(IOError, error)
+@loadfile(method=False, writable=True)
+def delete(filething):
+    """Completely removes the ID3 chunk from the AIFF file"""
+
+    try:
+        del RiffFile(filething.fileobj)[u'ID3']
```

Comment:
If the ID3v2 chunk should be considered as a multi purpose chunk (like RIFF and LIST) it should be in capitals everywhere. But that is a different discussion. ✔